config/config_params.go

```diff
@@ -317,6 +317,7 @@ func (config *Config) DatastoreConfig() api.CalicoAPIConfig {
 		if err != nil {
```

Comment:
Well, when I was playing with the k8s backend (prior to our chat earlier today), my complete invocation was:

	docker run --rm \
	-e CALICO_K8S_API_ENDPOINT=scale-test \
	-e FELIX_DATASTORETYPE=kubernetes \
	-e FELIX_LOGSEVERITYSCREEN=debug \
	calico/felix

In other words, if FELIX_DATASTORETYPE works to select the k8s datastore, you can then also use variables like CALICO_K8S_API_ENDPOINT to configure fields of the KubeConfig.  With the fix that I've proposed, that all seems to hang together and make sense.

I haven't yet noticed any doc or code based on DATASTORE_TYPE (without FELIX_ or CALICO_ prefix) - are you sure about that?